Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -120,10 +119,36 @@ function xASL_wrp_ProcessM0(x)
 
 			% Find the smallest of the previously selected positive nonzero TEs
 			minTE = min(uniqueNonzeroTE);
-			fprintf('Multi-TE M0 present. Using the shortest TE=%.1f ms for M0 calibration.\n', minTE*1000);
+			% JSON values were converted from BIDS (seconds) to Legacy (millisecond) during JSON reading

```

Comment:
OK, I will try to reduce the number of IF-statements, but we will need to have the if statements to report that we were choosing single volume from multiple. I'll try to also simplify this, but just saying that we won't be able to remove all IFs.

Reply:
I've checked the code, but it can't be really simplified to reduce the number of IF statements:
1. We need an IF to check if we have JSONM0
2. We need an IF to check presence of TE and TR field.
3. We need an IF again to check the presence of TR and TE fields to know if we update them.
4. So the only IF to get rid of potentially is the IF that checks if we have or haven't multiple values. But we need this IF anyway to report to the user that we found multiple values and are choosing a single one.

So we can't really simplify the code and I don't think this code is complicated. I've added extra comments.

Reply:
1. We need this now, but not when we go for BIDS-requirement (which we almost have)
2. Same as (1)
3. Same as (1)
But I agree.

Reply:
Yes - we will run later a full BIDS validator before running a pipeline later...